src/main/groovy/nu/studer/gradle/jooq/JooqGenerate.java

```diff
@@ -254,11 +264,20 @@ private void writeConfiguration(Configuration config, File file) {
         }
     }
 
-    private String xsdFileName() {
-        // use reflection to avoid inlining of the String constant org.jooq.Constants.XSD_CODEGEN
+    private String xsdResourcePath() {
         try {
-            Class<?> aClass = Class.forName("org.jooq.Constants");
-            return (String) aClass.getDeclaredField("XSD_CODEGEN").get(null);
+            // use reflection to avoid inlining of the String constant org.jooq.Constants.XSD_CODEGEN
+            Class<?> jooqConstants = Class.forName("org.jooq.Constants");
+
+            if (Arrays.stream(jooqConstants.getDeclaredFields())
+                    .map(Field::getName)
+                    .anyMatch(s -> s.equals("CP_CODEGEN"))) {
+                // jOOQ >= 3.12.0; we can use the "new" approach of constructing the XSD class path.
+                return (String) jooqConstants.getDeclaredField("CP_CODEGEN").get(null);
+            } else {
+                // For older versions, we resort to the previous hardwired approach.
+                return "/xsd/" + jooqConstants.getDeclaredField("XSD_CODEGEN").get(null);
+            }
```

Comment:
Tested with `3.15.1` (i.e. the version referenced by `build.gradle`) with great success. :tada: 

Note that the fix in this PR is not enough for full jOOQ 3.16 support; there is still the annoying `name()` issue (similar to, or perhaps even identical to, the one I reported in #204) that causes issues with jOOQ 3.16 even after this change is in place. I'll keep digging into that as well, but hope to submit a fix to that in a separate PR.

Reply:
(#208 fixes the `name()` issue btw)